iModelCore/ECDb/Tests/NonPublished/ECDbTests.cpp

```diff
@@ -561,5 +561,159 @@ TEST_F(ECDbTestFixture, NewFileECDbProfileVersion)
     ASSERT_EQ(m_ecdb.GetECDbProfileVersion(), expectedVersion);
     }
 
+class IssueListener : public ECN::IIssueListener
+    {
+    mutable bvector<Utf8String> m_issues;
+    void _OnIssueReported(ECN::IssueSeverity severity, ECN::IssueCategory category, ECN::IssueType type, Utf8CP message) const override
+        {
+        m_issues.push_back(message);
+        }
+    public:
+    Utf8StringCR GetLastError() const { return m_issues.back();}
+    void clearMessages() { m_issues.clear(); }
+    };
+
+TEST_F(ECDbTestFixture, TestGreatestAndLeastFunctionsWithLiterals)
+    {
+    IssueListener listener;
+    ASSERT_EQ(BE_SQLITE_OK, SetupECDb("newFile.ecdb"));
+    m_ecdb.AddIssueListener(listener);
+
+    // Test DQL statements
+    for (const auto& [lineNumber, sqlStatement, expectedStatus, expectedResult, errorMsg] : bvector<std::tuple<const int, const Utf8String, const ECSqlStatus, const std::vector<int>, const Utf8String>>
+        {
+        { __LINE__, "SELECT GREATEST()", ECSqlStatus::InvalidECSql, { 0 }, "Preparing the ECSQL 'SELECT GREATEST()' failed. Underlying SQLite statement failed to prepare: BE_SQLITE_ERROR wrong number of arguments to function MAX() (BE_SQLITE_ERROR) [SQL: SELECT MAX()]" },
+        { __LINE__, "SELECT GREATEST(4)", ECSqlStatus::Success, { 4 }, "" },
+        { __LINE__, "SELECT GREATEST(4, 2, 1, 5, 3)", ECSqlStatus::Success, { 5 }, "" },
+
+        { __LINE__, "SELECT LEAST()", ECSqlStatus::InvalidECSql, { 0 }, "Preparing the ECSQL 'SELECT LEAST()' failed. Underlying SQLite statement failed to prepare: BE_SQLITE_ERROR wrong number of arguments to function MIN() (BE_SQLITE_ERROR) [SQL: SELECT MIN()]" },
+        { __LINE__, "SELECT LEAST(4)", ECSqlStatus::Success, { 4 }, "" },
+        { __LINE__, "SELECT LEAST(4, 2, 1, 5, 3)", ECSqlStatus::Success, { 1 }, "" },
+        { __LINE__, "SELECT LEAST(4, 2, 1, 5, 3)", ECSqlStatus::Success, { 1 }, "" },
+
+        { __LINE__, "SELECT MAX()", ECSqlStatus::InvalidECSql, { 0 }, "Failed to parse ECSQL 'SELECT MAX()': syntax error" },
+        { __LINE__, "SELECT MAX(4)", ECSqlStatus::Success, { 4 }, "" },
+        { __LINE__, "SELECT MAX(4, 2, 1, 5, 3)", ECSqlStatus::InvalidECSql, { 0 }, "MAX function with multiple arguments isn't supported. Please use GREATEST instead." },
+        { __LINE__, "SELECT MAX(4, 2, 1, 5, 3)", ECSqlStatus::InvalidECSql, { 0 }, "MAX function with multiple arguments isn't supported. Please use GREATEST instead." },

```

Comment:
Use More complex examples with types and actual column and compute expressions in both SELECT and WHERE clause.

```
SELECT GREATEST(
       'a', 
       'b',
       'c')
```

```
SELECT LEAST(
       'a', 
       'b',
       'c')
```

```
SELECT LEAST(
       'a', 
       'b',
       'c')
```

```
SELECT MAX(MAX(3), MAX(4))
```